### PR TITLE
clean up example a smidge

### PR DESCRIPTION
**Pull request by colemickens**
Very minor changes...

1. Remove Xplat Client ID since it is treated differently than other client ids

2. Pull "Resource" up so it's easy to test different audience URLs.

2. Make forcibly refreshing the token optional

CC @ahmetalpbalkan

### CODE REVIEW

File: autorest/azure/example/main.go [hunk 1]
```diff
@@ -18,7 +18,9 @@ import (
 
 const resourceGroupURLTemplate = "https://management.azure.com/subscriptions/{subscription-id}/resourcegroups"
 const apiVersion = "2015-01-01"
-const xplatClientID = "04b07795-8ddb-461a-bbee-02f9e1bf7b46"
+
+const nativeAppClientID = "a87032a7-203c-4bf7-913c-44c50d23409a"
+const resource = azure.AzureresourceManagerScope
```
Comment (ahmetalpbalkan) [inline autorest/azure/example/main.go]: typo?

Comment (colemickens) [inline autorest/azure/example/main.go]: Yeah, I did `%s/Resource/resource/g` to fix a lint error and introduced a bunch of silliness. Travis is green again.....

**Comment by ahmetalpbalkan** [comment]
lgtm

**Comment by colemickens** [comment]
Thanks; merged.

**Pull request merged**